### ISSUE TITLE
Assertion errors during amalgmation

### ISSUE THREAD

**Issue by wd40bomber7** [vinniefalco/Amalgams#1]
Attempting to run a simple amalgamation results in a long list of assertion errors for no obvious reason. The output file looks like this:
![Capture](https://f.cloud.github.com/assets/2555227/190687/e2723462-7f06-11e2-80ec-c2f221da080c.PNG)

Notice how the header files stop immediately after the #pragma once and the rest of the file is ignored. 

Run log of the program with -v argument looks like this:
JUCE Assertion failure in c:\users\wd40bomber7\desktop\exciting\juce_core_amalga
m.cpp, line 1968
  Inlining     "Color.h"
JUCE Assertion failure in c:\users\wd40bomber7\desktop\exciting\juce_core_amalga
m.cpp, line 1968
  Inlining    "Vec.h"
name = <iostream>
JUCE Assertion failure in c:\users\wd40bomber7\desktop\exciting\juce_core_amalga
m.cpp, line 1968
  Inlining   "Window.h"
JUCE Assertion failure in c:\users\wd40bomber7\desktop\exciting\juce_core_amalga
m.cpp, line 1968
  Inlining    "Control.h"
JUCE Assertion failure in c:\users\wd40bomber7\desktop\exciting\juce_core_amalga
m.cpp, line 1968
  Inlining     "Brush.h"
JUCE Assertion failure in c:\users\wd40bomber7\desktop\exciting\juce_core_amalga
m.cpp, line 1968

I compiled my amalgamate binary with visual studio 2010 using the default settings provided by the project file.

**Comment by vinniefalco** [comment]
What is the exact command line that you are giving to run Amalgamate.exe? It looks like you need to use a full path in the arguments to the executable. I think that the assert is probably benign (its JUCE complaining that the path is not absolute). Try running the Release build.

Also, you can try building an amalgamation of JUCE using the JUCEAmalgam repo to verify that the amalgamate tool works.

**Comment by wd40bomber7** [comment]
The command I was running was:
a.exe -v Amalgamation.cpp Completed.cpp

(I had renamed the executable)
Compiling in release and using the same command except without renaming the executable ran without assertion errors. Furthermore upon more detailed examination of the Completed.cpp put out by the debug version (which threw the assertion errors) I noticed  that it actually executed correctly. So all in all, everything actually works and I should have selected release mode if I didn't want to see debug messages. No surprise there I suppose.Thank you!
